arangod/IResearch/IResearchLink.cpp

```diff
@@ -1241,17 +1280,42 @@ Result IResearchLink::init(
   const_cast<IResearchLinkMeta&>(_meta) = std::move(meta);
   _comparer.reset(_meta._sort);
 
+  // we should create stats for link only for Single Server or
+  // of for DB Server. But in case of DB Server we must check that
+  // link created for actual dataStore and not for ClusterInfo
+  if (ServerState::instance()->isSingleServer() || !clusterWideLink) {
+    _linkStats =
```

Comment:
maybe get a reference to `MetricsFeatures` once?

Reply:
Done